application/controllers/AuthenticationController.php

```diff
@@ -35,11 +41,32 @@ public function loginAction()
             $this->redirectNow(Url::fromPath('setup'));
         }
         $form = new LoginForm();
+        if (RememberMe::hasCookie()) {
+            $authenticated = false;
+            try {
+                $rememberMe = RememberMe::fromCookie();
+                $authenticated = $rememberMe->authenticate();
+                if ($authenticated) {
+                    $this->getResponse()->setCookie(
+                        $rememberMe->renewCookie()
```

Comment:
This should be `$rememberMe->renew()->getCookie()` where `renew()` just returns a new `RememberMe` instance. This way we have full flexibility. Maybe we have to fiddle with the newly created key pair and this way we're prepared.